Replicator/Replicator.cc

```diff
@@ -899,6 +899,7 @@ namespace litecore { namespace repl {
         _getCollectionsRequested = true;
         
         for (int i = 0; i < _collections.size(); i++) {
+            _sessionCollections.push_back(_collections[i].get());
```

Comment:
Why is this? Let me refer to static index as the index from _options. The active replicator always use the static index, right? And the static index will become protocol index. 
For the passive replicator, the static index does not have any bearing on the protocol. It only define the set that it is only willing to work with the client, ie collections of getCollections must be a subset. The request sends the mapping from index i to client spec, and the server must find the static index that corresponds to that client spec.
Right?

Reply:
This is for ease of use.  Now all we have to do to map is take the received collection index, and use it to retrieve the correct collection rather than say "if passive, use this and if active use that"

Reply:
Do we still populate all the workers, pusher/pullers, in the constructor of Replicator based on _options?